apps/aecore/lib/aecore/structures/data_tx.ex

```diff
@@ -0,0 +1,46 @@
+defmodule  Aecore.Structures.DataTx do
+  @moduledoc """
+  Aecore structure of a transaction data.
+  """
+
+  alias Aecore.Structures.DataTx
+  alias Aecore.Keys.Worker, as: Keys
+   @type t :: %DataTx{
+    type: atom(),
+    payload: any(),
+    from_acc: binary(),
+    fee: integer(),
+    nonce: non_neg_integer()
+   }
+
+  @doc """
+  Definition of Aecore DataT structure
+
+  ## Parameters
+  - type: To account is the public address of the account receiving the transaction
+  - payload:
+  - from_acc: From account is the public address of one account originating the transaction
+  - fee: The amount of a transaction
+  - nonce: A random integer generated on initialisation of a transaction.Must be unique
+
+  """
+  defstruct [:type, :payload, :from_acc , :fee, :nonce]
+  use ExConstructor
+
+#  @spec create(binary(), binary(), non_neg_integer(), non_neg_integer(), non_neg_integer(), non_neg_integer()) :: {:ok, DataTx.t()}
+  def create(type, payload, from_acc, fee, nonce) do
```

Comment:
@HristoHr @d-velev what do you think of this? I think it would be a great enhancement, when implemented correctly, as this can ease chain_state calculation a lot.

Reply:
@rdimitrovbg and me are willing to work on this issue and make our requested changes. We have other ideas as well continuing on the path of making the code more abstract and generic. @rdimitrovbg has a lot of experiece with working on big and abstract projects. He came up with the idea. Just to have in mind.